File: 2022/day-1/day-1.js
const input = require("fs").readFileSync(`${__dirname}/input.txt`).toString();

function sum(array) {
  return array.reduce((a, b) => a + b, 0);
}

let calorieSums = input
  .split("\n\n")
  .map((inventory) => {
    return sum(inventory.split("\n").map((str) => parseInt(str, 10)));
  })
  .sort((a, b) => b - a);

console.log(`Part 1: ${calorieSums[0]}`);
console.log(`Part 2: ${sum(calorieSums.slice(0, 3))}`);

File: 2022/day-1/day-1.py
input = open("day-1/input.txt", "r").read()

calorie_sums = []

for line in input.split("\n\n"):
    inventory = line.split("\n")
    calorie_sums.append(sum([int(str) for str in inventory]))

calorie_sums.sort(reverse=True)

print("Part 1: %d" % calorie_sums[0])
print("Part 2: %d" % sum(calorie_sums[0:3]))
